### ISSUE TITLE
Updating Collaborator. Original collaborator did not get updated. A new collaborator gets created, with updated info.

### ISSUE THREAD

**Issue by eugenelin89** [alistairliya/react#2]
1. Create a new business
2. Add a new collaborator and enter details 
5. Update. Reload.
5. Edit the collaborator (Change CFC code). Update. Reload.
Result:
- Original collaborator did not get updated.
- A new collaborator gets created, with updated info.

**Comment by eugenelin89** [comment]
Backend, views.py, line 625, in 
def edit_business:


```
for key in collaborator_data:
                # For each collaborator, need to figure out if it's a new collaborator or an existing one
                # if a business_user_id exists, then it's an existing collaborator
                # otherwisse it is a new collaborator
                collaborator = collaborator_data.get(key)
                print(f'\n??\nCollaborator {key}:\n {collaborator}')
                business_user_id = collaborator.get('id') # Exiting collaborator loaded directly from busi
``` 
collaborator object does not have an "id"

```
 {'advisor': {'id': 24, 'first_name': '', 'last_name': '', 'url': 'http://localhost:8000/api/users/24/', 'username': 'admin001', 'email': '', 'groups': [], 'created_businesses': ['http://localhost:8000/api/mybusiness/238/'], 'my_businesses': ['http://localhost:8000/api/businessuser/92/', 'http://localhost:8000/api/businessuser/93/', 'http://localhost:8000/api/businessuser/94/', 'http://localhost:8000/api/businessuser/109/', 'http://localhost:8000/api/businessuser/110/', 'http://localhost:8000/api/businessuser/114/', 'http://localhost:8000/api/businessuser/115/', 'http://localhost:8000/api/businessuser/117/', 'http://localhost:8000/api/businessuser/118/', 'http://localhost:8000/api/businessuser/120/', 'http://localhost:8000/api/businessuser/121/', 'http://localhost:8000/api/businessuser/122/', 'http://localhost:8000/api/businessuser/124/', 'http://localhost:8000/api/businessuser/126/', 'http://localhost:8000/api/businessuser/127/', 'http://localhost:8000/api/businessuser/128/', 'http://localhost:8000/api/businessuser/129/', 'http://localhost:8000/api/businessuser/130/', 'http://localhost:8000/api/businessuser/131/', 'http://localhost:8000/api/businessuser/132/', 'http://localhost:8000/api/businessuser/133/', 'http://localhost:8000/api/businessuser/134/', 'http://localhost:8000/api/businessuser/135/', 'http://localhost:8000/api/businessuser/136/', 'http://localhost:8000/api/businessuser/137/', 'http://localhost:8000/api/businessuser/138/', 'http://localhost:8000/api/businessuser/139/', 'http://localhost:8000/api/businessuser/140/', 'http://localhost:8000/api/businessuser/141/', 'http://localhost:8000/api/businessuser/142/', 'http://localhost:8000/api/businessuser/149/', 'http://localhost:8000/api/businessuser/150/', 'http://localhost:8000/api/businessuser/151/', 'http://localhost:8000/api/businessuser/152/', 'http://localhost:8000/api/businessuser/154/', 'http://localhost:8000/api/businessuser/155/', 'http://localhost:8000/api/businessuser/156/', 'http://localhost:8000/api/businessuser/157/', 'http://localhost:8000/api/businessuser/158/', 'http://localhost:8000/api/businessuser/159/', 'http://localhost:8000/api/businessuser/160/', 'http://localhost:8000/api/businessuser/161/', 'http://localhost:8000/api/businessuser/162/', 'http://localhost:8000/api/businessuser/163/', 'http://localhost:8000/api/businessuser/164/', 'http://localhost:8000/api/businessuser/165/']}, 'role': {'id': 1, 'user_role_name': 'owner', 'description': 'a user who is responsible for the business'}, 'cfcCode': '287-b', 'collaboratorStatus': {'id': 1, 'status_name': 'QT', 'description': 'Qualified Trainer'}, 'collaboratorPosition': {'id': 1, 'position_name': 'REP', 'description': 'REP'}, 'split': 0}
```

**Comment by eugenelin89** [comment]
Here is the issue.
In NBF8Advisor.js,
```
updateAdvisor(id, {advisor: selectedOption.value, role: role, cfcCode: cfcCode, collaboratorStatus: collaboratorStatus, collaboratorPosition: collaboratorPosition, split:split})
```
The actual ID from businessuser object isn't being stored in the json object in the parameter passed into updateAdvisor.

**Comment by eugenelin89** [comment]
Suggested Solution.
In useEffect in NBFF8Advisor, if we are loading an existing advisor, ie. in the scope of 
```
if(selectedAdvisors[id]){
```
save the "id" of the object extracted from selectedAdvisors[id] as a state variable. And then in the call to  updateAdvisor, include it it the json object.

**Comment by eugenelin89** [comment]
Solved half of the bug. Now, need to finish this in views.py
```
 if business_user_id:
                    print("\n EXISTING COLLABORATOR")
                    #print(f"\nExisting Collaborator: {collaborator} \n")
                    pass # HERE!!!!
                else:
```

**Comment by alistairliya** [comment]
The issue of creating a duplicate is resolved. However, after updating, changes are not saved.

**Comment by alistairliya** [comment]
Resolved in f16d975e2142183aae9fef39efda51c458960769 (backend)
